presto-kafka/src/main/java/io/prestosql/plugin/kafka/BaseKafkaConsumerFactory.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.prestosql.plugin.kafka;
+
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+
+import java.util.Properties;
+
+public abstract class BaseKafkaConsumerFactory
```

Comment:
I dislike this class. We generally avoid subclassing as a means to share code (part of the "prefer delegation" from Effective Java), and here we aren't saving anything, since it's trivial to implement `KafkaConsumerFactory` directly. The primary purpose seems to be to prevent subclasses from creating `KafkaConsumer` directly, but I don't see a reason for that.